### DISCUSSION TITLE
Events and arrays with mqtt

### DISCUSSION THREAD

**Discussion by 13** [arendst/Tasmota#21358] [Q&A] [answered]
### PROBLEM DESCRIPTION
I'm not able to trigger ENERGY.Power[1] over mqtt.

I tried without success following 2 approaches:
```
Subscribe PowerTotal, tasmota/tele/tasmota_5FF8B2/SENSOR, ENERGY.Power
Rule3 ON Event#PowerTotal[1] DO var1 %value% ENDON

Subscribe PowerTotal, tasmota/tele/tasmota_5FF8B2/SENSOR, ENERGY
Rule3 ON Event#PowerTotal#Power[1] DO var1 %value% ENDON
```

```
tasmota/tele/tasmota_5FF8B2/SENSOR
{"Time":"2024-05-05T13:27:21","ENERGY":{"TotalStartTime":"2024-05-01T11:36:59","Total":31.043,"Yesterday":7.154,"Today":3.322,"TodaySumImport":1.141,"TodaySumExport":0.001,"ExportActive":[0.000,0.000,0.000],"Power":[11,0,0],"ApparentPower":[181,0,0],"ReactivePower":[181,0,0],"Factor":[0.06,0.00,0.00],"Frequency":[50,49,53],"Voltage":[231,0,0],"Current":[0.785,0.013,0.013],"CurrentNeutral":0.008}}
```

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x ] Tasmota binary firmware version number used: 13.4.0
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
The combo of `Subscribe` and `Event` commands only supports simple values, not arrays. You can get it working with the Berry language on ESP32, or by using rules on the source device to publish a simpler payload to pick up.
